utils/src/main/java/io/neow3j/contract/NeoURI.java

```diff
@@ -0,0 +1,184 @@
+package io.neow3j.contract;
+
+import io.neow3j.utils.AddressUtils;
+import io.neow3j.utils.Strings;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class NeoURI {
+
+    private String uri;
+
+    protected NeoURI(Builder builder) {
+        this.uri = builder.uri;
+    }
+
+    public String getUri() {
+        return uri;
+    }
+
+    public static class Builder {
+        private String scheme;
+        private String address;
+
+        private Map<String, String> query;
+        private List<String> hashs;
+        private List<String> remarks;
+
+        private String uri;
+
+        public Builder() {
+            this.scheme = "neo";
+            this.query = new HashMap<>();
+            this.hashs = new ArrayList<>();
+            this.remarks = new ArrayList<>();
+        }
+
+        public Builder toAddress(String address) {
+            this.address = address;
+            return this;
+        }
+
+        public Builder asset(String asset) {
+            this.query.put("asset", asset);
+            return this;
+        }
+
+        public Builder amount(String amount) {
+            // TODO: 20.04.20 Michael: handle decimals if needed.
+            this.query.put("amount", amount);
+            return this;
+        }
+
+        public Builder amount(int amount) {
+            this.query.put("amount", String.valueOf((double) amount));
```

Comment:
This conversion of types here is a bit dangerous.
Can you use `Integer` instead? Then you could do: `Integer.toString(amount);`